sklearn/utils/_plotting.py

```diff
@@ -250,6 +251,125 @@ def _validate_curve_kwargs(
         return curve_kwargs_
 
 
+class _LineTooltipMixin:
+    """Mixin class to add a tooltip to a line in a plot.
+
+    The tooltip displays 2 to 3 informations: the x value, the y value and an optional
+    t value for parametric curves (x(t), y(t)).
+    """
+
+    def _add_line_tooltip(self, *, x_label, y_label, t_label=None, t_vals=None):
+        """Create the line tooltip and connect it to a mouse hover event.
+
+        Parameters
+        ----------
+        x_label : str
+            The label for the x value.
+
+        y_label : str
+            The label for the y value.
+
+        t_label : str, default=None
+            The label for the parameter value.
+
+        t_vals : list of ndarrays, default=None
+            The parameter values along the line.
+        """
+        self._x_label_short = x_label
+        self._y_label_short = y_label
+        self._t_label = t_label
+
+        self.line_tooltip_ = self.ax_.annotate(
+            text="",
+            xy=(0, 0),
+            xytext=(20, 20),
+            textcoords="offset points",
+            fontsize="small",
+            bbox=dict(boxstyle="round", fc=(0.8, 0.8, 0.8, 0.8)),
+            arrowprops=dict(arrowstyle="-"),
+            zorder=10,  # bring to front
+        )
+        self.line_tooltip_.set_visible(False)
+
+        # Set an attribute on the axes annotation to be able to keep only one visible
+        # at a time when there are multiple display instances that share an axes.
+        setattr(self.line_tooltip_, "_skl_line_tooltip", True)
+
```

Comment:
This is not ideal, we set an attribute on a matplotlib object. An alternative could be to have a class attribute that is a mapping from axes to something and reuse the same matplotlib annotation if a new display instance reuses the axes. But it's not very clean either...

Reply:
> won't be a proper matplotlib.axes.Axes instance, so still not ideal

Why do you think that would be a problem? Subclassing sounds a bit cleaner, unless the matplotlib documentation explicitly discourages subclassing for one reason or another.